lib/tools/stats.js

```diff
@@ -0,0 +1,17 @@
+var convert = require('./convert_time');
+var BigInt = require('./bigint');
+
+module.exports = function(file) {
+  return {
+    birthtime: convert(BigInt.fromBuffer(file.CreationTime).toNumber()),
+    atime: convert(BigInt.fromBuffer(file.LastAccessTime).toNumber()),
+    mtime: convert(BigInt.fromBuffer(file.LastWriteTime).toNumber()),
+    ctime: convert(BigInt.fromBuffer(file.ChangeTime).toNumber()),
+    size: BigInt.fromBuffer(file.EndofFile).toNumber(),
+    isDirectory: function() {
+      if (typeof file.FileAttributes === 'number')
+        return file.FileAttributes === 16;
+      return BigInt.fromBuffer(file.FileAttributes).toNumber() === 16;
```

Comment:
Do the conversion before so that the test is not duplicated.

Reply:
sorry, do you want me to convert int to Buffer?

Reply:
No no, the other way around, like you did:
```suggestion
      var attr = file.FileAttributes;
      if (typeof attr !== 'number') {
        attr = BigInt.fromBuffer(attr).toNumber();
      }
      return attr === 16;
```

Reply:
ok clear, thx